homework/python_hw2.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Oct 14 18:41:44 2019
+
+@author: pedro
+"""
+
+#Import packages
+import pandas as pd
+import psycopg2
+import os
+
+#Coonect to database hw1
+conn = psycopg2.connect("dbname=hw1 user=postgres host=localhost")
+cur = conn.cursor()
+
+#Read tables from operational database
+cur.execute("SELECT * FROM customers;")
+cust_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+cur.execute("SELECT * FROM offices;")
+office_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+cur.execute("SELECT * FROM employees;")
+emp_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+cur.execute("SELECT * FROM products;")
+prod_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+cur.execute("SELECT * FROM order_info;")
+oinfo_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+cur.execute("SELECT * FROM order_products;")
+oprod_data = pd.DataFrame(cur.fetchall(), columns = [desc[0] for desc in cur.description])
+
+#Create facts table
+facts1 = pd.merge(oprod_data, oinfo_data, on ='order_number')
+facts2 = pd.merge(facts1, prod_data[['product_code','buy_price']], on = 'product_code')
+facts = pd.merge(facts2, cust_data[['customer_number','sales_rep_employee_number']], on = 'customer_number')
+facts['profit'] = facts['price_each'] - facts['buy_price']
+
+#Create dimension tables
+def create_date_table(start='2000-01-01', end='2020-12-31'):
```

Comment:
Nice little function, but I'm not sure I like hardcoding a fixed range. It solves a problem (the creation of date ID's), but it's not very flexible (what's going to happen in 2020!) -- remember, this is the kind of thing that led to havoc in the year 2000! We can do better than hardcoding end dates for our software ;)